op-node/rollup/derive/batches.go

```diff
@@ -139,7 +148,14 @@ func CheckBatchEspresso(cfg *rollup.Config, log log.Logger, l2SafeHead eth.L2Blo
 		log.Warn("error reading suggested L1 origin", "err", err, "l1 head", jst.Next.L1Head)
 		return BatchUndecided
 	}
-	expectedL1Origin := EspressoL1Origin(cfg, l2SafeHead, suggestedL1Origin)
+	nextL1Number := l2SafeHead.L1Origin.Number + 1
+	nextL1Block, err := l1.L1BlockRefByNumber(context.Background(), nextL1Number)
+	if err != nil {
+		log.Warn("error reading next possible L1 origin", "err", err, "origin", nextL1Number)
+		return BatchUndecided
+	}
+	nextL1BlockEligible := nextL1Block.Time <= cfg.BlockTime+l2SafeHead.Time
```

Comment:
It might be cleaner to encapsulate this logic in `EspressoL1Origin`, so it just takes the next `L1BlockRef`, not a `bool`. Then there's no chance of bugs from different callers having a different idea of what "eligible" means.

Reply:
agreed this feels better